### PR TITLE
Edit contact

### PR DESCRIPTION
**Pull request by Inisi**
finished with edit contact functionality

### CODE REVIEW

File: src/components/exercises/App1/AddContact.js [hunk 1]
```diff
@@ -1,32 +1,59 @@
...
+    setEmail("");
   };
 
   return (
     <section className="add-contact">
-        <input type="text" placeholder={'Enter Name'} value={name} onChange={(e) => setName(e.target.value)}/>
-        <input type="text" placeholder={'Enter Email'} value={email} onChange={(e) => setEmail(e.target.value)}/>
+      <input
+        type="text"
+        placeholder={isEditForm ? props.contact.name : "Enter Name"}
```
Comment (jurikaca) [inline src/components/exercises/App1/AddContact.js]: Do not change placeholder, you can leave same value for placeholder.

File: src/components/exercises/App1/AddContact.js [hunk 2]
```diff
@@ -1,32 +1,59 @@
...
+        value={name}
```
Comment (jurikaca) [inline src/components/exercises/App1/AddContact.js:46]: Make value dynamic when it exists

File: src/components/exercises/App1/AddContact.js [hunk 3]
```diff
@@ -1,32 +1,59 @@
...
+      <input
+        type="text"
+        placeholder={isEditForm ? props.contact.email : "Enter Email"}
+        value={email}
+        onChange={(e) => setEmail(e.target.value)}
+      />
       <br />
-      <button onClick={addContactHandler}>Add new contact</button>
+      {!isEditForm && (
+        <button onClick={addContactHandler}>Add new contact</button>
```
Comment (jurikaca) [inline src/components/exercises/App1/AddContact.js]: Make the logic only for text, not the whole button

File: src/components/exercises/App1/redux/slices/contactSlice.js [hunk 4]
```diff
@@ -1,54 +1,92 @@
...
+      contactSlice.caseReducers.setDefaultSelectedContact(state);
+    },
+    editContact: (state, action) => {
+      state.data.forEach((contact) => {
+        if (contact.id === action.payload) {
+          contact.edit = !contact.edit;
+        }
+      });
+
+      state.updateRequired = true;
```
Comment (jurikaca) [inline src/components/exercises/App1/redux/slices/contactSlice.js]: remove this

File: src/components/exercises/App1/redux/slices/contactSlice.js [hunk 5]
```diff
@@ -1,54 +1,92 @@
...
+    data: [],
+    selectedContact: null,
+    updateRequired: false,
+  },
+  reducers: {
+    addContact: (state, action) => {
+      state.data.push(action.payload);
+    },
+    updateContact: (state, action) => {
+      const contactId = action.payload;
```
Comment (jurikaca) [inline src/components/exercises/App1/redux/slices/contactSlice.js]: - remove unnecessary variables
- use .forEach function to loop contacts
- remove console

**Review by jurikaca — CHANGES_REQUESTED** [review]

**Comment by Inisi** [comment]
Po ti paske qen i fulliqt

**Review by jurikaca — APPROVED** [review]

**Pull request merged**